src/components/store/SizeSelector/index.tsx

```diff
@@ -0,0 +1,42 @@
+import styles from '@/components/store/SizeSelector/style.module.scss';
+import { PublicMerchItemOption } from '@/lib/types/apiResponses';
+import { Fragment, useId } from 'react';
+
+interface SizeSelectorProps {
+  currSize: string | undefined;
+  uuid: string;
+  options: PublicMerchItemOption[];
+  onSizeChange: (currSize: string) => void;
+}
+
+const SizeSelector = ({ currSize, options, onSizeChange, uuid }: SizeSelectorProps) => {
+  const myID = useId();
+
+  const myOptions = options.map(val => {
+    const tempId = val.metadata?.value ? `${val.metadata.value}${myID}` : `${val.uuid}${myID}`;
+    return (
+      <>
+        <input
+          id={tempId}
+          name="state-d"
+          type="radio"
+          defaultChecked={currSize === val.metadata?.value}
+          onClick={() => onSizeChange(val.metadata?.value)}
+        />
+        <label htmlFor={tempId}>{val.metadata?.value}</label>
+      </>
+    );
+  });
+
+  return (
+    <div className={styles.sizeSelector}>
+      {options.length > 1 ? (
+        <Fragment key={uuid}>
+          <h4>Size</h4>
+          <form className={styles.switch}>{myOptions}</form>
+        </Fragment>
+      ) : null}
+    </div>
+  );
```

Comment:
What I meant for adding the `key` prop was to add it to elements inside arrays, like `myOptions`, so React can tell the items apart. So below in the suggestion, I added `key` to the `Fragment` inside `map` since it creates an array.

```suggestion
  const myOptions = options.map(val => {
    const tempId = val.metadata?.value ? `${val.metadata.value}${myID}` : `${val.uuid}${myID}`;
    return (
      <Fragment key={tempId}>
        <input
          id={tempId}
          name="state-d"
          type="radio"
          defaultChecked={currSize === val.metadata?.value}
          onClick={() => onSizeChange(val.metadata?.value)}
        />
        <label htmlFor={tempId}>{val.metadata?.value}</label>
      </Fragment>
    );
  });

  return (
    <div className={styles.sizeSelector}>
      {options.length > 1 ? (
        <>
          <h4>Size</h4>
          <form className={styles.switch}>{myOptions}</form>
        </>
      ) : null}
    </div>
  );
```

The fragment at the bottom doesn't need a `key` because React knows that there will it either be one element or none